src/core/public/http/http_service.test.ts

```diff
@@ -19,35 +19,213 @@
+
+    fetchMock.get('*', { foo: 'bar' });
+
+    const json = await http.fetch('/my/path');
+
+    expect(json).toEqual({ foo: 'bar' });
+  });
+
+  it('should prepend url with basepath by default', async () => {
+    const { http } = setupService();
+
+    fetchMock.get('*', {});
+    await http.fetch('/my/path');
+
+    expect(fetchMock.lastUrl()).toBe('http://localhost/myBase/my/path');
+  });
+
+  it('should not prepend url with basepath when disabled', async () => {
+    const { http } = setupService();
+
+    fetchMock.get('*', {});
+    await http.fetch('my/path', { prependBasePath: false });
+
+    expect(fetchMock.lastUrl()).toBe('/my/path');
+  });
+
+  it('should make request with defaults', async () => {
+    const { http } = setupService();
+
+    fetchMock.get('*', {});
+    await http.fetch('/my/path');
+
+    expect(fetchMock.lastOptions()!).toMatchObject({
+      method: 'GET',
+      credentials: 'same-origin',
+      headers: {
+        'Content-Type': 'application/json',
+        'kbn-version': 'kibanaVersion',
+      },
+    });
+  });
+
+  it('should reject on network error', async () => {
+    const { http } = setupService();
+
+    expect.assertions(1);
+    fetchMock.get('*', { status: 500 });
+
+    await expect(http.fetch('/my/path')).rejects.toThrow(/Internal Server Error/);
+  });
+
+  it('should contain error message when throwing response', async () => {
+    const { http } = setupService();
+
+    fetchMock.get('*', { status: 404, body: { foo: 'bar' } });
+
+    await expect(http.fetch('/my/path')).rejects.toMatchObject({
+      message: 'Not Found',
+      body: {
+        foo: 'bar',
+      },
+      response: {
+        status: 404,
+        url: 'http://localhost/myBase/my/path',
+      },
+    });
+  });
+
+  it('should support get() helper', async () => {
+    const { http } = setupService();
+
+    fetchMock.get('*', { method: 'POST' });
+    await http.get('/my/path');
+
+    expect(fetchMock.lastOptions()!.method).toBe('GET');
+  });
+
+  it('should support post() helper', async () => {
+    const { http } = setupService();
+
+    fetchMock.post('*', {});
+    await http.post('/my/path', { method: 'GET', body: '{}' });
+
+    expect(fetchMock.lastOptions()!.method).toBe('POST');
+  });
+
+  it('should support put() helper', async () => {
+    const { http } = setupService();
+
+    fetchMock.put('*', {});
+    await http.put('/my/path', { method: 'GET', body: '{}' });
+
+    expect(fetchMock.lastOptions()!.method).toBe('PUT');
+  });
+
+  it('should support del() helper', async () => {
+    const { http } = setupService();
+
+    fetchMock.delete('*', {});
+    await http.del('/my/path', { method: 'GET' });
+
+    expect(fetchMock.lastOptions()!.method).toBe('DELETE');
+  });
+
+  it('should return an abortable promise', () => {
+    const { http } = setupService();
+
+    fetchMock.get('*', {});
+
+    const abortable = http.fetch('/my/path');
+
+    expect(typeof abortable.then).toBe('function');
```

Comment:
optional: why not just check `toBeInstanceOf(Promise)` explicitly?

Reply:
Good idea. This was migrated from the `kfetch` test.